dev/source/docs/sitl-with-airsim.rst

```diff
@@ -0,0 +1,281 @@
+.. _sitl-with-airsim:
+
+======================
+Using SITL with AirSim
+======================
+
+.. youtube:: -WfTr1-OBGQ
+   :width: 100%
+
+`AirSim <https://github.com/microsoft/AirSim>`__ is a simulator for drones, cars and more, built on Unreal Engine (they also have an experimental Unity release, but right now it's hasn't been implemented with ArduPilot)
+
+It is open-source, cross platform and provides excellent physically and visually realistic simulations. It has been developed with a puprose of becoming a platform for AI research to experiment with deep learning, computer vision and reinforcement learning algorithms for autonomous vehicles. 
+
+Currently, only support for ArduCopter vehicle has been developed in AirSim & ArduPilot
+
+Installing AirSim
+=================
+
+`Binaries <https://microsoft.github.io/AirSim/docs/use_precompiled/>`__ are available for AirSim, but since this is a very recent development, you'll have to build it yourself (on Windows as well as Linux).
+
+Build on Windows
+----------------
+
+The main page for Windows setup is `here <https://github.com/microsoft/AirSim/blob/master/docs/build_windows.md>`__, but the instructions have been described below as well. Check the FAQs on the page if there are any problems.
+
+Install Unreal Engine
++++++++++++++++++++++
+
+#. `Download <https://www.unrealengine.com/download>`__ the Epic Games Launcher.
+
+#. Run the Epic Games Launcher, open the Library tab on the left pane.
+
+   Click on the Add Versions which should show the option to download Unreal 4.18. If you have multiple versions of Unreal installed then make sure 4.18 is set to current by clicking down arrow next to the Launch button for the version.
+
+Build AirSim
+++++++++++++
+
+#. Install Visual Studio 2017.
+
+   Make sure to select VC++ and Windows SDK 8.1 while installing VS 2017.
+
+#. Start x64 Native Tools Command Prompt for VS 2017.
+
+#. Clone the repo
+
+    ::
+
+        git clone https://github.com/Microsoft/AirSim.git
+
+#. Currently, the `AirSim PR <https://github.com/microsoft/AirSim/pull/2075>`__  for ArduCopter Support hasn't been merged, so you'll need to fetch that particular branch for using Ardupilot.
+
+    ::
+    
+        cd AirSim
+        git remote add <remote-name> https://github.com/rajat2004/AirSim.git
+        git fetch <remote-name>
+        git checkout -b <your-branch-name> <remote-name>/pr-arducopter
```

Comment:
I'm not sure if this the best way to go about this. Can someone suggest a more suitable method?

Reply:
Another possibility is to have something like ArduPilot's own version of AirSim, since the AirSim repo seems to be very much inactive these days (something similar to this was there for earlier for JSBSim I think?)

Reply:
yes, if it doesn't get merged soon we will fork into ardupilot repo